tutorials/navigation/navigation_optimizing_performance.rst

```diff
@@ -0,0 +1,116 @@
+.. _doc_navigation_optimizing_performance:
+
+Optimizing Navigation Performance
+=================================
+
+.. image:: img/nav_optimization.webp
+
+Common Navigation related performance problems can be categorized into the following topics:
+
+- Performance problems with parsing SceneTree nodes for navigation mesh baking.
+- Performance problems with baking the actual navigation mesh.
+- Performance problems with NavigationAgent path queries.
+- Performance problems with the actual path search.
+- Performance problems with synchronizing the navigation map.
+
+In the following sections information can be found how to identify and fix or at least mitigate their impact on framerate.
+
+Performance problems with parsing SceneTree nodes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tip::
+
+    Prefer using simple shapes with as few edges as possible e.g. nothing rounded like a circle, sphere or torus.
+    
+    Prefer using physics collision shapes over complex visual meshes as source geometry as meshes need to be received from the GPU and are commonly overdetailed.
+
+In general avoid using very complex geometry as source geometry for baking navigation meshes.
+E.g. never use a very detailed visual mesh as parsing its shape to data arrays and voxelizing it for the navigation mesh baking will take ages for no real quality gain on the final navigation mesh.
+Instead, use a very simplified level of detail version of a shape. Even better, use very primitive shapes like boxes and rectangles that only roughly cover the same geometry but still yield a baked result good enough for pathfinding.
+
+Prefer using simple physics collision shapes over visual meshes as the source geometry for baking navigation meshes.
+Physics shapes are by default very limited and optimized shapes that are easy and quick to parse. A visual mesh on the other hand can range from simple to insanely complex.
+On top, to gain access to visual mesh data the parser needs to request the mesh data arrays from the RenderingServer as visual mesh data is stored directly on the GPU.
+This requires locking the RenderingServer thread and can severely impact framerate at runtime while the rendering runs multi-threaded.
+If the rendering runs single-threaded the framerate impact might be even worse and the mesh parsing might freeze the entire game for a few seconds on complex meshes.
+
+Performance problems with navigation mesh baking
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tip::
+
+    At runtime always prefer to use a background thread for baking navigation meshes.
+    
+    Increase NavigationMesh ``cell_size`` and ``cell_height`` to create less voxels.
+    
+    Change the ``SamplePartitionType`` from watershed to monotone or layers to gain baking performance.
```

Comment:
May make sense to mark watershed/monotone/layers somehow, maybe italics or `` too.